services/src/main/java/org/keycloak/services/clientpolicy/executor/SecureRedirectUriEnforceExecutor.java

```diff
@@ -56,6 +57,44 @@ public String getProviderId() {
         return componentModel.getProviderId();
     }
 
+    @Override
+    public void beforeRegister(ClientPolicyContext context) throws ClientPolicyException {
+        if (context instanceof AdminClientRegisterContext || context instanceof DynamicClientRegisterContext) {
+            confirmSecureRedirectUris(((ClientUpdateContext)context).getProposedClientRepresentation().getRedirectUris());
+        } else {
+            throw new ClientPolicyException(OAuthErrorException.INVALID_REQUEST, "not allowed input format.");
+        }
+    }
+
+    @Override
```

Comment:
We might not add methods for existing client registration polices. Instead of it, we could treat the re-implement client registration policies in the client policy manner, not the existing client registration manner.

We could not modify the existing merged conditions and executors when introducing the new events (in this case, `REGISTERED`, `UPDATED`, `VIEW` and `UNREGISTER`).